src/com/google/enterprise/adaptor/CommandStreamParser.java

```diff
@@ -467,6 +467,19 @@ public void readFromRetriever(DocId docId, Response response) throws IOException
     command = readCommand();
     while (command != null) {
       switch (command.getOperation()) {
+        case CONTENT:
+        case UP_TO_DATE:
+        case NOT_FOUND:
+          // Finish by putting accumulated ACL into response.
+          if (sendAclWithDocument) {
+            aclBuilder.setPermits(permits);
+            aclBuilder.setDenies(denies);
+            response.setAcl(aclBuilder.build());
+          }
+          break;
+        default:
+      }
+      switch (command.getOperation()) {
```

Comment:
I pondered doing so initially, so I was tempted to say sure. However, this class calls command.getOperation() multiple times after readCommand() in four separate places. I changed the comment a little bit for clarity. Another option would be put this code into a setAcl or finishAcl helper method, and instead of this extra switch statement, call the ACL helper separately in the corresponding cases in the existing switch statement.